EA-10: Allow device alias symlinks

fdgonthier opened the pull request:
Added a new configuration instruction in ueventd to make alias symlinks
to specific devices in /dev. The instruction uses the USB vendorId and
productId field as well as the major/minor of the device.

Change-Id: I0b670aeb4358b75ebc55c011589def9ab50e2f62

fdgonthier committed:
EA-10: Allow device alias symlinks

Added a new configuration instruction in ueventd to make alias symlinks
to specific devices in /dev. The instruction uses the USB vendorId and
productId field as well as the major/minor of the device.

Change-Id: I0b670aeb4358b75ebc55c011589def9ab50e2f62

fdgonthier committed:
EA-10: Remove debugging statement

Change-Id: I7d9fe8fdc2cf30c4eaf73092b818e420a714e669

fdgonthier committed:
EA-10: Remove the link when the device is removed

Change-Id: I282b42f83fb9844407433e6f532d3dd991ac1987

ajasmin commented on init/devices.cpp:
```diff
@@ -458,7 +471,65 @@ void DeviceHandler::HandleUevent(const Uevent& uevent) {
+
+    parent_dir = Dirname("/sys" + upath);
+
+    while ((parent_dir != "/" && parent_dir != ".")) {
```
How can we ever reduce the path to "." given that it starts with "/sys"?

ajasmin commented on init/devices.cpp:
```diff
@@ -458,7 +471,65 @@ void DeviceHandler::HandleUevent(const Uevent& uevent) {
+
+        if (match_productId && match_vendorId && match_major && match_minor) {
+            alias_link = alias.AliasTo();
+            return true;
```
Does one alias per (productId, vendorId, major, minor) tuple cover the case of the modem which exposes a number of tty?

I haven't looked at this in details.

ajasmin commented:
Well done. This integrates well with the existing code and contains tests.

Now we'll need the corresponding config.

fdgonthier committed:
EA-10: Allow symlink masks with minor number or bInterfaceNumber

This is an improvement on the previous implementation as it does not
force the major/minor to be specified if it's not needed. Only the
Vendor/Product IDs are really required. If not present, then a symlink
will be created for each of the endpoints that has to be created for
the USB device. To avoid name conflicts, the alias name can include %i
or %m has mask. %i will be replaced by the bInterfaceNumber of the
endpoint, and %m will be replaced by the minor number.

Change-Id: I3f75c68664a39860279c8d5b64d6cfa6062c8e00

fdgonthier committed:
EA-10: Stop when we reach /sys instead of /

Change-Id: I3e5bc5e9b90cf431989a8d92f088c572ded13b4c
